corehq/apps/userreports/views.py

```diff
@@ -439,6 +441,28 @@ def preview_data_source(request, domain, config_id):
     return render(request, "userreports/preview_data.html", context)
 
 
+def process_url_params(params, column_descriptions):
+    columns = {column['name']: column for column in column_descriptions}
+    keyword_filters = {}
+    sql_filters = []
+    for key, value in params.items():
+        if key == 'format':
+            continue
+        column = columns.get(key, None)
+        if not column:
+            raise BadSpecError('Invalid filter parameter: {}'.format(key))
+
+        if (
+            value == 'last30'
+            and isinstance(column['type'], (types.Date, types.DateTime))
+        ):
+            expr = column['expr']
+            sql_filters.append(expr.between('2014-02-02', '2014-10-02'))
+        else:
+            keyword_filters[key] = value
+    return keyword_filters, sql_filters
```

Comment:
Oh yeah, I should totally pull `format` into this.
I'm not sure about converting params to sql filters - this is done like `q = q.filter_by(**keyword_filters)` below.  In order to convert it to something you can pass to `filter`, I think you'd have to get it from the column object and use SQLAlchemy's operator overloading - maybe something like `getattr(expr, key) == val`
I think I'll start with just a `namedtuple` and see how it goes.